### PR TITLE
Add fromBuffer and toBuffer to PrivateKey

### LINKED ISSUE #739: PrivateKey `fromBuffer` G7
- Add `fromBuffer` static method to PrivateKey - [G7](https://github.com/bitpay/bitcore/blob/v0.8/CONTRIBUTING.md#g7---general-standard-methods)

### PR DESCRIPTION
**Pull request by eordano**
Closes #739

### CODE REVIEW

**Comment by braydonf** [comment]
In what situation is a BN is a buffer? Usually most BN's are an instance of BN or a hexa.

**Comment by eordano** [comment]
Is it ok with you if I change the current `fromBuffer` (which takes a DER encoded key) to something like `fromDERBuffer` instead of a `fromBNBuffer`? The use case I'm thinking of is `new PrivateKey(get256RandomBitsAsBuffer())`.

I'd like to keep as much versatility in the constructor as possible, because otherwise a new user needs to read through the doc to remember what was the particular method's name that he needed to use.

**Comment by braydonf** [comment]
The current `toBuffer` is the BN buffer (I forgot about that). So `fromBuffer` should accept the output from that.

File: lib/privatekey.js [hunk 1]
```diff
@@ -282,7 +282,17 @@ PrivateKey.isValid = function(data, network){
...
-}
+};
+
+/**
+ * Returns a private key generated by a big number serialized into a buffer
+ *
+ * @param {Buffer} arg
+ * @returns {PrivateKey}
+ */
+PrivateKey.fromBuffer = function(buffer, arg0, arg1) {
```
Comment (braydonf) [inline lib/privatekey.js]: `arg0` I believe is network, and arg1 wouldn't be used.

**Comment by maraoz** [comment]
ACK

**Pull request merged**